WalletWasabi/Wallets/Wallet.cs

```diff
@@ -196,22 +196,54 @@ public List<TransactionSummary> BuildHistorySummary(bool sortForUI = false)
 	/// <summary>
 	/// Gets the wallet transaction with the given txid, if the transaction exists.
 	/// </summary>
-	public bool TryGetTransaction(uint256 txid, [NotNullWhen(true)] out SmartTransaction? smartTransaction)
+	public bool TryGetTransactionSummary(uint256 txid, [NotNullWhen(true)] out TransactionSummary? transactionSummary)
 	{
-		// The lock is necessary to make sure that coins registry and transaction store do not change in this code block.
-		// The assumption is that the transaction processor is the only component modifying coins registry and transaction store.
 		lock (TransactionProcessor.Lock)
 		{
-			smartTransaction = null;
-			bool isKnown = Coins.IsKnown(txid);
-
-			if (isKnown && !BitcoinStore.TransactionStore.TryGetTransaction(txid, out smartTransaction))
+			if (TryGetTransactionNoLock(txid, out SmartTransaction? smartTransaction))
 			{
-				throw new UnreachableException($"{nameof(Coins)} and {nameof(BitcoinStore.TransactionStore)} are not in sync (txid '{txid}').");
+				if (!Coins.TryGetTxAmount(txid, out Money? money))
+				{
+					throw new UnreachableException($"{nameof(Coins)} should contain money for '{txid}'.");
```

Comment:
Is this really something unreachable?

Would it be more precise if we would throw an `ArgumentException($"{nameof(Coins)}" doesn't know about '{txid}')`, or something similar to this?

Reply:
OK, makes sense.